modules/couchbase/src/test/java/org/testcontainers/couchbase/CouchbaseContainerTest.java

```diff
@@ -1,22 +1,64 @@
+/*
+ * Copyright (c) 2020 Couchbase, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.testcontainers.couchbase;
 
-import org.junit.Assert;
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.document.JsonDocument;
+import com.couchbase.client.java.document.json.JsonObject;
+import com.couchbase.client.java.env.CouchbaseEnvironment;
+import com.couchbase.client.java.env.DefaultCouchbaseEnvironment;
 import org.junit.Test;
 
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
 public class CouchbaseContainerTest {
 
     @Test
-    public void shouldUseCorrectDockerImage() {
-        CouchbaseContainer couchbaseContainer = new CouchbaseContainer().withClusterAdmin("admin", "foobar");
+    public void testBasicContainerUsage() {
+        BucketDefinition bucketDefinition = new BucketDefinition("mybucket");
 
-        Assert.assertEquals(CouchbaseContainer.DOCKER_IMAGE_NAME + CouchbaseContainer.VERSION,
-            couchbaseContainer.getDockerImageName());
-    }
+        try (CouchbaseContainer container = new CouchbaseContainer().withBucket(bucketDefinition)) {
+            container.start();
 
-    @Test
-    public void shouldStopWithoutThrowingException() {
-        CouchbaseContainer couchbaseContainer = new CouchbaseContainer();
-        couchbaseContainer.start();
-        couchbaseContainer.stop();
+            CouchbaseEnvironment environment = DefaultCouchbaseEnvironment
+                .builder()
+                .bootstrapCarrierDirectPort(container.getMappedPort(11210))
+                .bootstrapHttpDirectPort(container.getMappedPort(8091))
```

Comment:
Here I was wondering if it wouldn't be nicer to make those default ports public constants in the container class?

Reply:
@kiview one other option would be to provide explicit getters for only those two that you might need for bootstrapping? So people don't have to concern themselves with the implementation details at all. This is needed for SDK 2 only, for SDK 3 all you need to do is getConnectionString() .. so something like getCarrierDirectPort() and getHttpDirectPort() and the same for the ssl ports if someone wants to use that.